tests/unit/test_extractor.py

```diff
@@ -0,0 +1,172 @@
+import pytest
+import pandas as pd
+from cognite.client.data_classes import EventWrite
+from cdf_fabric_replicator.extractor import CdfFabricExtractor
+from cdf_fabric_replicator.config import Config
+from cognite.extractorutils.base import CancellationToken
+from cognite.extractorutils.statestore import LocalStateStore
+from cognite.extractorutils.configtools import load_yaml
+
+@pytest.fixture(scope="session")
+def config_raw():
+    yield """
+    # Logging configuration
+    logger:
+        console:
+            level: INFO
+
+    # Cognite project to stream your datapoints from
+    cognite:
+        host: https://api.cognitedata.com
+        project: unit_test_extractor
+
+        idp-authentication:
+            token-url: https://get-a-token.com/token
+            client-id: abc123
+            secret: def567
+            scopes:
+                - https://api.cognitedata.com/.default
+        extraction-pipeline:
+            external-id: test-fabric-extractor
+
+    #Extractor config
+    extractor:
+        state-store:
+            local:
+                path: test_states.json
+        subscription-batch-size: 10000
+        ingest-batch-size: 100000
+        poll-time: 5
+
+    source:
+        abfss_event_table_path: source_table_path
+        abfss_directory: file_path
+        abfss_raw_time_series_table_path: /table/path
+        data_set_id: 123456
+
+    destination:
+        type: test_events
+        time_series_prefix: test_ts_
+    """
+@pytest.fixture(scope="session")
+def config(config_raw):
+    yield load_yaml(config_raw, Config)
+
+
+@pytest.fixture(scope="session")
+def extractor(config):
+    stop_event = CancellationToken()
+    extractor = CdfFabricExtractor(stop_event=stop_event)
+    extractor.config = config
+    extractor.client = extractor.config.cognite.get_cognite_client("test_extractor")
+    extractor.state_store = LocalStateStore(extractor.config.extractor.state_store.local.path)
+    
+    yield extractor
+
+
+def test_parse_abfss_url(extractor):
+    url = 'https://container@account.dfs.core.windows.net/path'
+    result = extractor.parse_abfss_url(url)
+    assert result == ('container', 'account', '/path')
+
+
+def test_write_time_series_to_cdf(extractor, mocker):
+    data = {
```

Comment:
It looks like this test data is used in several tests, can you define this as a fixture?

Reply:
It's actually different based on the ids provided etc. there is one method other than this one that uses the exact same data set dict and I think it helps to see the data that should be filtered etc when doing those test cases for readability.

Reply:
I see, at the least the timestamps and values seem to be the same so is there any way to consolidate those?